Library/Homebrew/formulary.rb

```
@@ -123,7 +123,16 @@ def initialize(bottle_name)
     end
 
     def get_formula(spec, alias_path: nil)
-      formula = super
+      bottle_version = Utils::Bottles.resolve_version @bottle_filename
+      path = "#{name}/#{bottle_version}/.brew/#{name}.rb"
+      contents = Utils.popen_read "tar", "-xO", "-f", @bottle_filename,
+        "--strip-components", "3", path, err: :close
```

I don't want `err: :close` to be going throughout the codebase when they are no-ops for most users and when, in this case, the error should never be hit. I'm tempted to say rather than falling back this should raise an exception instead. Thoughts?

---

An error is expected when used with a bottle that does not contain `.brew/#{name.rb}`, as for bottles older than about a year so. For these bottles without an internal formula, the behaviour of `brew install` remains unchanged.

---

Does it make more sense to change the warning to an error and get rid of the old code path?

---

Agreed. I'll change it.